src/annotations/components/annotationForm/AnnotationForm.tsx

```diff
@@ -0,0 +1,182 @@
+// Libraries
+import React, {ChangeEvent, FC, FormEvent, useContext, useReducer} from 'react'
+
+// Components
+import {
+  Overlay,
+  Button,
+  ComponentColor,
+  Form,
+  Grid,
+  Columns,
+  InputRef,
+  TextAreaRef,
+  ButtonType,
+} from '@influxdata/clockface'
+import AnnotationSummaryInput from 'src/annotations/components/annotationForm/AnnotationSummaryInput'
+import AnnotationTimeStartInput from 'src/annotations/components/annotationForm/AnnotationTimeStartInput'
+import AnnotationTimeStopInput from 'src/annotations/components/annotationForm/AnnotationTimeStopInput'
+import AnnotationMessageInput from 'src/annotations/components/annotationForm/AnnotationMessageInput'
+import AnnotationTypeToggle from 'src/annotations/components/annotationForm/AnnotationTypeToggle'
+import AnnotationStreamSelector from 'src/annotations/components/annotationForm/AnnotationStreamSelector'
+
+// Form State
+import {
+  annotationReducer,
+  getInitialAnnotationState,
+  AnnotationType,
+  annotationFormIsValid,
+  Annotation,
+  getAnnotationFromDraft,
+} from 'src/annotations/reducers/annotationReducer'
+
+// Types
+import {AnnotationStream} from 'src/annotations/constants/mocks'
+
+// Contexts
+import {OverlayContext} from 'src/overlays/components/OverlayController'
+
+interface Props {
+  title: 'Edit' | 'Add'
+  type: AnnotationType
+  timeStart: string
+  timeStop: string
+  summaryText?: string
+  messageText?: string
+  streamID?: string
+  onSubmit: (Annotation: Annotation) => void
+}
+
+const AnnotationForm: FC<Props> = ({
+  title,
+  type,
+  timeStart,
+  timeStop,
+  summaryText,
+  messageText,
+  streamID,
+  onSubmit,
+}) => {
+  const initialAnnotationState = getInitialAnnotationState(
+    type,
+    timeStart,
+    timeStop,
+    summaryText,
+    messageText,
+    streamID
+  )
+
+  const [state, dispatch] = useReducer(
+    annotationReducer,
+    initialAnnotationState
+  )
+
+  const {onClose} = useContext(OverlayContext)
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>): void => {
+    e.preventDefault()
+
+    const formIsValid = annotationFormIsValid(state, dispatch)
+
+    if (formIsValid) {
+      onSubmit(getAnnotationFromDraft(state))
+      onClose()
+    }
+  }
+
+  const handleSummaryChange = (e: ChangeEvent<InputRef>): void => {
+    dispatch({type: 'updateSummary', payload: e.target.value})
```

Comment:
these actions should follow the convention used in the rest of the app. but before you do that, why have individual actions for every property on the store and not have a single action that takes a Partial<StateShape> ?